modules/repository/temp.go

```diff
@@ -15,30 +16,28 @@ import (
 
 // LocalCopyPath returns the local repository temporary copy path.
 func LocalCopyPath() string {
-	if filepath.IsAbs(setting.Repository.Local.LocalCopyPath) {
-		return setting.Repository.Local.LocalCopyPath
+	if setting.Repository.Local.LocalCopyPath == "" {
+		return filepath.Join(setting.TempPath, "local-repo")
+	} else if !filepath.IsAbs(setting.Repository.Local.LocalCopyPath) {
+		return filepath.Join(setting.TempPath, setting.Repository.Local.LocalCopyPath)
 	}
-	return filepath.Join(setting.AppDataPath, setting.Repository.Local.LocalCopyPath)
+	return setting.Repository.Local.LocalCopyPath
 }
 
 // CreateTemporaryPath creates a temporary path
-func CreateTemporaryPath(prefix string) (string, error) {
+func CreateTemporaryPath(prefix string) (string, context.CancelFunc, error) {
 	if err := os.MkdirAll(LocalCopyPath(), os.ModePerm); err != nil {
 		log.Error("Unable to create localcopypath directory: %s (%v)", LocalCopyPath(), err)
-		return "", fmt.Errorf("Failed to create localcopypath directory %s: %w", LocalCopyPath(), err)
+		return "", func() {}, fmt.Errorf("failed to create localcopypath directory %s: %w", LocalCopyPath(), err)
 	}
 	basePath, err := os.MkdirTemp(LocalCopyPath(), prefix+".git")
 	if err != nil {
 		log.Error("Unable to create temporary directory: %s-*.git (%v)", prefix, err)
-		return "", fmt.Errorf("Failed to create dir %s-*.git: %w", prefix, err)
+		return "", func() {}, fmt.Errorf("failed to create dir %s-*.git: %w", prefix, err)
```

Comment:
Here and above, it could just return `nil` instead of `func() {}` for CancelFunc, because the callers should have checked `err`

Reply:
b88c3a6af3